src/vm/eventpipe.cpp

```diff
@@ -245,90 +256,70 @@ EventPipeSessionID EventPipe::Enable(
     }
     CONTRACTL_END;
 
+    // If the state or arguments are invalid, bail here.
+    if (sessionType == EventPipeSessionType::File && strOutputPath == nullptr)
+        return 0;
+    if (sessionType == EventPipeSessionType::IpcStream && pStream == nullptr)
+        return 0;
+
     EventPipeSessionID sessionId;
-    EventPipe::RunWithCallbackPostponed(
-        [&](EventPipeProviderCallbackDataQueue* pEventPipeProviderCallbackDataQueue)
-        {
-            // Create a new session.
-            SampleProfiler::SetSamplingRate(static_cast<unsigned long>(profilerSamplingRateInNanoseconds));
-            EventPipeSession *pSession = s_pConfig->CreateSession(
-                sessionType,
-                circularBufferSizeInMB,
-                pProviders,
-                numProviders);
-
-            // Enable the session.
-            sessionId = Enable(strOutputPath, pSession, sessionType, pStream, pEventPipeProviderCallbackDataQueue);
-        }
-    );
+    RunWithCallbackPostponed([&](EventPipeProviderCallbackDataQueue *pEventPipeProviderCallbackDataQueue) {
+        // FIXME: What's the story here? SampleProfiler should be enabled only once.
+        SampleProfiler::SetSamplingRate(static_cast<unsigned long>(profilerSamplingRateInNanoseconds));
+
+        // TODO: Maybe "rundownEnabled" param should be user input.
```

Comment:
Thanks for thinking about it - I think we want to make sure it is a user input.
https://github.com/dotnet/diagnostics/issues/189

Reply:
Maybe we also want to control the sampling rate (currently hardcoded).

Reply:
It will be challenging to let the users control the sampling rate because if two sessions both want to control it who wins? I think we should find out what AI profiler sets it to now and assuming that is a reasonable default we should hardcode that number. Its not a great story, but I think it is an OK one for this release.

Reply:
I'm thinking we can leave this for when we have the protocol ready. I will open an issue later.